Source/MonoGame.Extended/Maps/Tiled/TiledMap.cs

```diff
@@ -83,6 +100,68 @@ public TiledObjectGroup CreateObjectGroup(string name, TiledObject[] objects, bo
             return objectGroup;
         }
 
+        public TiledMap Build()
+        {
+            var tileVertices = new List<VertexPositionTexture>();
+            var tileIndexes = new List<short>();
+            var tileLayers = _layers.OfType<TiledTileLayer>();
```

Comment:
Sort the layers here by Z value to fix the depth buffer issue. You want layers which are closer to the screen to processed last.